src/components/routes/homepage/EventForm.js

```diff
@@ -0,0 +1,134 @@
+import React, { useState } from 'react';
+import {
+  Button,
+  Container,
+  FormControl,
+  InputLabel,
+  Input,
+  TextField,
+  Typography,
+  makeStyles
+} from '@material-ui/core';
+import { EventTimePicker } from './EventTimePicker';
+import { ImagePicker } from './ImagePicker';
+import { MediaCard } from './MediaCard';
+import { Modal } from './Modal';
+import { validateEvent } from '../../../functions/validateEvent';
+import imagePlaceholder from '../../../assets/no_image_placeholder.png';
+
+const useStyles = makeStyles(theme => ({
+  formContainer: {
+    minHeight: '100vh',
+    paddingTop: theme.spacing(10),
+    textAlign: 'center',
+    display: 'grid',
+    gridTemplateColumns: 'repeat(2, 50%)',
+    alignItems: 'center',
+    justifyContent: 'center',
+    [theme.breakpoints.down(800)]: {
+      minHeight: 'max-content',
+      display: 'unset'
+    }
+  },
+  input: {
+    margin: `${theme.spacing(2)}px 0px`
+  },
+  timepickersContainer: {
+    margin: theme.spacing(2),
+    padding: `${theme.spacing(1)}px 0px`,
+    display: 'grid',
+    gridTemplateColumns: '45% 10% 45%'
+  },
+  submitButton: {
+    margin: theme.spacing(5),
+    gridColumn: '1 / span 2',
+    justifySelf: 'center',
+    color: '#fff',
+    backgroundColor: theme.palette.primary.light,
+    '&:hover': {
+      backgroundColor: theme.palette.primary.main
+    }
+  }
+}));
+
+export const EventForm = ({ onFormSubmit }) => {
+  const [title, setTitle] = useState('');
+  const [start, setStart] = useState(new Date());
+  const [end, setEnd] = useState(new Date());
+  const [venue, setVenue] = useState('');
+  const [description, setDescription] = useState('');
+  const [imageBase64, setImageBase64] = useState(imagePlaceholder);
+  const [modal, setModal] = useState({
+    isOpen: false,
+    title: ''
+  });
+
+  const classes = useStyles();
+
+  const event = {
+    title,
+    start,
+    end,
+    venue,
+    image_url: imageBase64,
```

Comment:
should just use `imageBase64` throughout all event creation components for better clarity, essentially there isn't a single "url" involved anywhere

Reply:
so the problem here is since `event` is passed as a prop of `<MediaCard />` right, in `<MediaCard />` component, you have that `image_url` thing. Because client will directly call `https://rcltech-raven.s3...` and fetch the image

Reply:
I see. `MediaCard` is used in two places, the `Events` rendered in homepage, and `EventForm`. In that case, take `event` image as `imageBase64` and do a spread operator `{...event, image_url: event.imageBase64}` when passed to `MediaCard` may be better? It feels more helpful in terms of using the `createEvent` mutation later on for consistency here.

Reply:
ahh I see, so will have both `imageBase64` and `image_url` fields rightt?? will do that in a moment